nx/source/services/set.c

```diff
@@ -686,3 +686,69 @@ Result setsysGetFatalDirtyFlags(u64 *flags_0, u64 *flags_1) {
 
     return rc;
 }
+
+Result setsysGetDeviceNickname(char* buffer) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    ipcAddRecvBuffer(&c, buffer, 0x80, BufferType_Normal);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 77;
+
+    Result rc = serviceIpcDispatch(&g_setsysSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result setsysSetDeviceNickname(const char* buffer) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    ipcAddSendBuffer(&c, buffer, strlen(buffer) + 1, BufferType_Normal);
```

Comment:
Can you copy `buffer` to a tmp 0x80-byte buf, then use that for ipcAddSendBuffer?

Reply:
Like this? I followed the style from setsysGetSettingsItemValueSize. Should I also rename `buffer` to `nickname` and change `nickname` (the tmp buffer) to `send_nickname` for style consistency with how some of the other commands do it?

```c
// in set.h
#define SET_MAX_NICKNAME_SIZE 0x80

// in set.c
Result setsysSetDeviceNickname(const char* buffer) {    
    char nickname[SET_MAX_NICKNAME_SIZE];
    memset(nickname, 0, SET_MAX_NICKNAME_SIZE);
    strncpy(nickname, buffer, SET_MAX_NICKNAME_SIZE - 1);
    
    IpcCommand c;
    ipcInitialize(&c);
    ipcAddSendBuffer(&c, nickname, SET_MAX_NICKNAME_SIZE, 0);

    struct {
        u64 magic;
        u64 cmd_id;
    } *raw;

    raw = ipcPrepareHeader(&c, sizeof(*raw));

    raw->magic = SFCI_MAGIC;
    raw->cmd_id = 78;

    Result rc = serviceIpcDispatch(&g_setsysSrv);

    if (R_SUCCEEDED(rc)) {
        IpcParsedCommand r;
        ipcParse(&r);

        struct {
            u64 magic;
            u64 result;
        } *resp = r.Raw;

        rc = resp->result;
    }

    return rc;
}
```

Reply:
yeah